lib/adapter.js

```diff
@@ -155,6 +162,7 @@ class MochaAdapter {
         if (err) {
             this.runner.lastError = err
         }
+        process.emit(event, message)
```

Comment:
Here's the issue. The runner adds command hooks which enable events to be sent up to the BaseReporter for `runner:beforecommand` and `runner:beforecommand`. It also sends `runner:command`, `runner:result` and `runner:screenshot`. Each of these events needs to have enough information in the payload for the BaseReporter to be able to identify exactly which CID, spec file and test the event belongs to, otherwise we can't add the event data to the output for a test in, for example, the allure reporter.

As some events come from the mocha framework, and some from the runner, I couldn't find any way to get the test title to the runner other than by messaging it with `process.emit`. However, I've limited this to `test:start` in https://github.com/webdriverio/wdio-mocha-framework/commit/e332fc1e5e7972d752704d00b5444db3bfc4adce, and added a note to explain it.

If you can find another way of doing this, I'd be very happy!